src/pages/SearchPage/SearchPage.test.js

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import { Route, MemoryRouter } from 'react-router-dom';
+import {
+  render, fireEvent, cleanup, waitForElementToBeRemoved,
+} from '@testing-library/react';
+import axiosMock from 'axios';
+import { DEFAULT_SUBREDDIT, DEFAULT_PATH } from '../../config';
+import { dummyPosts } from '../../dummyData';
+import Theme, { theme } from '../../styles/theme';
+import SearchPage, { parseRedditData } from './SearchPage';
+import '@testing-library/jest-dom/extend-expect';
+import 'jest-styled-components';
+
+const ONE_YEAR_IN_SECONDS = 60 * 60 * 24 * 365;
+const NUM_GRID_ITEMS = 7 * 24; // days * hours
+
+// Avoiding mutating global variables (beforeEach/afterEach) for reasons
+//    argued here: https://kentcdodds.com/blog/avoid-nesting-when-youre-testing
+
+jest.mock('axios');
+afterEach(() => {
+  afterEach(cleanup);
+  jest.clearAllMocks();
+});
+
+const renderWithRouter = (component, path, page) => {
+  const url = `/${path}/${page}`;
+  return (
+    render(
+      <MemoryRouter initialEntries={[url]}>
+        <Theme>
+          <Route path={`/${path}/:subreddit`}>{component}</Route>
+        </Theme>
+      </MemoryRouter>,
+    )
+  );
+};
+
+function setup(mockState = 'pass', data = []) {
+  const oneYearAgo = Math.round(new Date().getTime() / 1000) - ONE_YEAR_IN_SECONDS;
+  const url = `https://api.pushshift.io/reddit/search/submission/?${DEFAULT_PATH}=${DEFAULT_SUBREDDIT}&sort=desc&sort_type=score&after=${oneYearAgo}&size=500`;
+  if (mockState === 'pass') {
+    axiosMock.get.mockResolvedValueOnce({ data });
+  } else if (mockState === 'fail') {
+    axiosMock.get.mockRejectedValue({ data });
+  }
+  const { getByTestId, getAllByTestId, queryByTestId } = renderWithRouter(
+    <SearchPage />, DEFAULT_PATH, DEFAULT_SUBREDDIT,
+  );
+  return {
+    getByTestId,
+    getAllByTestId,
+    queryByTestId,
+    url,
+  };
+}
+
+function getHeatmapValues(dataArray) {
+  const parsedData = parseRedditData(dataArray);
+  return parsedData.map((day) => day.map((hour) => hour.length)).flat();
+}
+
+describe('Search controls', () => {
+  test('Default input is used when search page is loaded', async () => {
+    const { getByTestId } = setup();
+    const input = getByTestId('search-input');
+    expect(input).toBeInTheDocument();
```

Comment:
You don't need to add extra assertions `toBeInTheDocument` when you use `getBy*` or `findBy*`. Those will fail if the element is not there

Reply:
The same goes for other occurrences of `toBeInTheDocument` here

Reply:
I think this would still be worth refactoring. It's pretty obvious to someone who is experienced with testing that this can be simplified